packages/account-sdk/src/core/communicator/Communicator.ts

```diff
@@ -129,8 +142,29 @@ export class Communicator {
       })
       .then(() => {
         if (!this.popup) throw standardErrors.rpc.internal();
-        logPopupSetupCompleted();
+        logPopupSetupCompleted(this.preference.mode);
         return this.popup;
       });
+
+    // Add timeout for embedded mode
+    if (this.preference.mode === 'embedded') {
+      const IFRAME_TIMEOUT = 10000;
+      const timeoutPromise = new Promise<never>((_, reject) => {
+        setTimeout(() => reject(new Error('iframe_timeout')), IFRAME_TIMEOUT);
+      });
+
+      return Promise.race([popupLoadedPromise, timeoutPromise]).catch((error) => {
```

Comment:
I resorted to a timeout because browsers don't allow you to inspect errors that iframes throw. So there's no way to know if it failed to load or if it threw a specific error.